notebooks/DATB/psf_grid_simulation_photometry/GriddedPSFModel_simulation_and_photometry.ipynb

```diff
@@ -0,0 +1,471 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Creating a simulated starfield image and performing photometry using `webbpsf` and `photutils.GriddedPSFModel`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The simulated image will be made using a spatially-dependent JWST point-spread function (PSF) generated by `webbpsf`.  `photutils` will be then used to perform PSF-fitting photometry on the simulated JWST NIRCam image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from astropy.io import fits\n",
+    "from astropy.modeling.fitting import LevMarLSQFitter\n",
+    "from astropy.stats import gaussian_sigma_to_fwhm\n",
+    "from astropy.table import Table\n",
+    "from astropy.visualization import simple_norm\n",
+    "\n",
+    "from webbpsf.gridded_library import display_psf_grid\n",
+    "from webbpsf.utils import to_griddedpsfmodel\n",
+    "\n",
+    "from photutils.datasets import make_noise_image\n",
+    "from photutils import BasicPSFPhotometry\n",
+    "from photutils import DBSCANGroup, MMMBackground"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We start by loading into FITS HDU a pre-generated grid of JWST PSFs for NIRCam detector A1 in F090W."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = 'https://stsci.box.com/shared/static/6h8wsr2ts0t24s79cxmnyk8bldt0vv3i.fits'\n",
+    "hdu = fits.open(url)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now create a `photutils.GriddedPSFModel` from the FITS HDU."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "psfmodel = to_griddedpsfmodel(hdu, ext=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "psfmodel"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`psfmodel` is an instance of a `photutils.GriddedPSFModel`.  Let's view some of it's attributes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The oversampled PSF grid is stored internally as a 3D `numpy.ndarray`.  Here there are 16 PSFs (from a 4x4 reference grid) and the shape of each is 404x404 pixels."
```

Comment:
Might be good to clarify what the various dimensions *are* (or link to webbpsf docs if that's easier and they exist)

Reply:
OK, I can point to this:

https://photutils.readthedocs.io/en/stable/api/photutils.psf.GriddedPSFModel.html#photutils.psf.GriddedPSFModel

Reply:
I added some text explaining the dimensions.